src/components/Avatar/Avatar.svelte

```diff
@@ -0,0 +1,92 @@
+<script lang="ts">
+    import type { Snippet } from 'svelte';
+    import type { Action } from 'svelte/action';
+
+    interface AvatarProps {
+        href?: string;
+        action?: Action<HTMLElement, string>;
+        actionParams?: string;
+
+        // Root (figure)
+        base?: string;
+        background?: string;
+        text?: string;
+        fontWeight?: string;
+        width?: string;
+        border?: string;
+        rounded?: string;
+        shadow?: string;
+        classes?: string;
+
+        // Image
+        imageSrc?: string;
+        imageAlt?: string;
+        imageBase?: string;
+        imageRest?: string;
+
+        // Snippets
+        children?: Snippet;
+    }
+
+    let {
+        href = "",
+        action = () => {},
+        actionParams = "",
+
+        // Root (figure)
+        base = "overflow-hidden isolate",
+        background = "bg-surface-400-500-token",
+        width = "w-16",
+        border = "",
+        rounded = "rounded-full",
+        shadow = "",
+        classes = "",
+
+        // Image
+        imageBase = "w-full object-cover",
+        imageSrc = "",
+        imageAlt = "",
+        imageRest = "",
+
+        // Snippets
+        children,
+    } = $props<AvatarProps>();
+</script>
+
+{#snippet figure()}
+    <figure class="{base} {background} {width} {border} {rounded} {shadow} {classes}">
+        {#if children}
+        <!-- TODO: when Snippet types are implemented in svelte-5 check for type and if string provide an SVG with the text/initials -->
+            {@render children()}
+            <!-- TODO: investigate a better approach using Tailwind. -->
+            <!-- <svg class="{initialsBase} {initialsRest}" viewBox="0 0 512 512">
```

Comment:
Feel free to nix the comments here. We have the v2 implementation for reference if needed.